app/models/reservation.rb

```diff
@@ -110,44 +118,25 @@ def self.unique_equipment_items?(reservations)
     item_ids == item_ids.uniq
   end
 
-  def self.number_for_model_on_date(date, model_id, source)
-    # count the number of reservations that overlaps a date within
-    # a given array of source reservations and that matches
-    # a specific model id
-    number_for(date, model_id, source, :equipment_model_id)
+  # Counts the number of reservations in source that overlap with the given date
+  # If no date is given, defaults to today
+  # If attrs are given, only counts the reservation if it has the
+  # specified attributes
+  def self.number_for(source, date: Time.zone.today, **attrs)
+    source.to_a.count { |r| r.overlaps_with(date) && r.attrs?(attrs) }
   end
 
-  def self.number_for_category_on_date(date, category_id, reservations)
-    number_for(date, category_id, reservations, :category_id)
+  # Same as number_for, just over a range of dates
+  def self.number_for_date_range(source, date_range, **attrs)
+    date_range.map { |d| Reservation.number_for(source, date: d, **attrs) }
   end
 
-  def self.number_for(date, value, source, property)
-    count = 0
-    source.each do |r|
-      if r.start_date <= date && r.due_date >= date &&
-         r.send(property) == value
-        count += 1
-      end
-    end
-    count
-  end
+  ## Getter style instance methods ##
 
-  def self.number_overdue_for_eq_model(model_id, reservations)
-    # count the number of overdue reservations for a given
-    # eq model out of an array of source reservations
-    #
-    # used in rendering the catalog in order to save db queries
-    #
-    # 0 queries
-    count = 0
-    reservations.each do |r|
-      count += 1 if r.overdue && r.equipment_model_id == model_id
-    end
-    count
+  def active?
```

Comment:
Is this used anywhere? Searching across the diff seems to indicate that it wasn't...